runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewDelegate.java

```diff
@@ -145,13 +153,73 @@ private static void init(final Context context) {
             CommandLine.init(readCommandLine(context.getApplicationContext()));
         }
 
-        ResourceExtractor.setMandatoryPaksToExtract(MANDATORY_PAKS);
-        int resListResId = context.getResources().getIdentifier(
+        final boolean isSharedMode = !context.getPackageName().equals(
+                context.getApplicationContext().getPackageName());
+
+        HashMap<String, ResourceEntry> resourceList = new HashMap<String, ResourceEntry>();
+        if (isSharedMode) {
+            for (String resource : MANDATORY_PAKS) {
+                resourceList.put(resource, new ResourceEntry(0, "", resource));
+            }
+        } else {
+            int resListId = context.getResources().getIdentifier(
                 XWALK_RESOURCES_LIST_RES_NAME, "array", context.getClass().getPackage().getName());
-        if (resListResId == 0) {
-            resListResId = context.getResources().getIdentifier(
+            if (resListId == 0) {
+                resListId = context.getResources().getIdentifier(
                     XWALK_RESOURCES_LIST_RES_NAME, "array", context.getPackageName());
+            }
+
+            try {
```

Comment:
If `resListId` is 0, we will catch `NotFoundException` later and still make the APK crash. Since the resources in this list is mandatory, it's not allowed to go without anyone of them